clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2841,23 +2838,22 @@ class KernelCallOperatorVisitor
     }
     return true;
   }
+
+  CXXMethodDecl *GetCallOperator() {
```

Comment:
Is there a reason you are not checking if you can use the CallOperator saved previously?